### PR TITLE
Update developer focussed documentation

### PR DESCRIPTION
**Pull request by sholderbach**
- CONTRIBUTING guide
- Update Readme and lib.rs
- remove old checklist

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Contributing
+
+reedline's development is currently primarily driven by the [nushell project](https://github.com/nushell) to provide its indigenous interactive REPL and explore options for pleasant interaction with a shell and programming language.
```
Comment (jntrnr) [inline CONTRIBUTING.md:3]: remove "indigenous" and starting with "interactive REPL" reads a bit better.

**Review by jntrnr — COMMENTED** [review]
just found a little wording thing, otherwise seems good

File: CONTRIBUTING.md (hunk 1 above)
Comment (sholderbach) [reply CONTRIBUTING.md:3]: Like?
```
reedline's development is primarily driven by the [nushell project](https://github.com/nushell) at the moment to provide its interactive REPL and explore options for pleasant interaction with a shell and programming language.
```

Comment (sholderbach) [reply CONTRIBUTING.md]: I split that wordy sentence :D

**Pull request merged**